src/megaclient.cpp

```diff
@@ -4662,8 +4665,23 @@ bool MegaClient::procsc()
 
                     if (!insca_notlast)
                     {
+                        sc_checkSequenceTag(string());
                         app->catchup_result();
                     }
+
+                    if (pendingsccommit && sctable && !reqs.cmdsInflight())
```

Comment:
Yes I agree, and in fact I started to do that a long time back, but then realized that we are not updating User related commands, they still update our data structures on command response, not in actionpackets (and don't have the cmdseq to tie them together) - and User data is also committed to the sctable along with nodes, matching the SCSN.   So, I don't think we can make that nice simplification unless we decide to also update the User based commands in the API.

Reply:
You're fully correct. Let's keep it by now. In case we update Users (and also PCRs and Chats) to v3 at some point, we can proceed with the simplification. Thanks